### PR TITLE
Normalize special headers in playback record matcher

### LINKED ISSUE #11569: [BUG] Recorded tests that include header values with a semicolon delimited string are not matched properly
**Describe the bug**
Recorded tests that include header values with a semicolon delimited string are not matched properly.

ex: `request.Headers.Add("Content-Type", "application/json;odata=nometadata");`

**Expected behavior**
Recorded header values match as expected.

**Actual behavior (include Exception or Stack Trace)**

System.InvalidOperationException : Unable to find a record for the request POST https://chrissprim.table.core.windows.net/Tables?$format=application%2Fjson%3Bodata%3Dfullmetadata
    Header differences:
        <Content-Type> values differ, request <application/json;odata=nometadata>, record <application/json; odata=nometadata>

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Record a test with a header as described
2. Attempt to run the recorded test in playback mode

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/11569

### CODE REVIEW

File: sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs [hunk 1]
```diff
@@ -171,26 +178,47 @@ private string JoinHeaderValues(string[] values)
...
         private int CompareHeaderDictionaries(SortedDictionary<string, string[]> headers, SortedDictionary<string, string[]> entryHeaders, HashSet<string> ignoredHeaders, StringBuilder descriptionBuilder = null)
         {
             int difference = 0;
             var remaining = new SortedDictionary<string, string[]>(entryHeaders, entryHeaders.Comparer);
             foreach (KeyValuePair<string, string[]> header in headers)
             {
-                if (remaining.TryGetValue(header.Key, out string[] values))
+                var requestHeaderValues = header.Value;
+                var headerName = header.Key;
+
```
Comment (christothes) [inline sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs:200]: Could we just `continue` here if `ignoredHeaders.Contains(headerName)` and remove the two checks for this below?

File: sdk/core/Azure.Core/tests/RecordSessionTests.cs [hunk 2]
```diff
@@ -200,6 +201,29 @@ public void SavingRecordingSanitizesValues()
...
+            // Use HttpClientTransport as it does header normalization
+            var originalRequest = new HttpClientTransport().CreateRequest();
+            originalRequest.Method = RequestMethod.Get;
+            originalRequest.Uri.Reset(new Uri("http://localhost"));
+            originalRequest.Headers.Add(name, "application/json;odata=nometadata");
+
+            var playbackRequest = new MockTransport().CreateRequest();
+            playbackRequest.Method = RequestMethod.Get;
+            playbackRequest.Uri.Reset(new Uri("http://localhost"));
+            playbackRequest.Headers.Add(name, "application/json;odata=nometadata");
```
Comment (christothes) [inline sdk/core/Azure.Core/tests/RecordSessionTests.cs:219]: Should we add at least one ignored header also to validate that logic?

**Review by christothes — APPROVED** [review]
LGTM - just a couple minor suggestions

File: sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs (hunk 1 above)
Comment (pakrym) [reply sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs:200]: Let me see, you might be right.

File: sdk/core/Azure.Core/tests/RecordSessionTests.cs (hunk 2 above)
Comment (pakrym) [reply sdk/core/Azure.Core/tests/RecordSessionTests.cs:219]: I think I'm missing something. Why do we need an ignored header here?

Comment (christothes) [reply sdk/core/Azure.Core/tests/RecordSessionTests.cs:219]: Just to ensure that they are, in fact, ignored in the re-normalize logic.

**Pull request merged**